src/ctapipe/reco/impact.py

```diff
@@ -2,68 +2,73 @@
 """
 
 """
-import math
+import copy
+from string import Template
 
 import numpy as np
 import numpy.ma as ma
 from astropy import units as u
 from astropy.coordinates import AltAz, SkyCoord
 from iminuit import Minuit
-from scipy.optimize import least_squares, minimize
 from scipy.stats import norm
 
-from ctapipe.containers import (
-    ReconstructedEnergyContainer,
-    ReconstructedGeometryContainer,
-)
-from ctapipe.coordinates import (
+from ctapipe.core import traits
+
+from ..containers import ReconstructedEnergyContainer, ReconstructedGeometryContainer
+from ..coordinates import (
+    CameraFrame,
     GroundFrame,
     NominalFrame,
     TiltedGroundFrame,
     project_to_ground,
 )
-from ctapipe.core import Component
-from ctapipe.image import mean_poisson_likelihood_gaussian, neg_log_likelihood
-from ctapipe.instrument import get_atmosphere_profile_functions
-from ctapipe.utils.template_network_interpolator import (
+from ..core import Provenance
+from ..fitting import lts_linear_regression
+from ..image.cleaning import dilate
+from ..image.pixel_likelihood import (
+    mean_poisson_likelihood_gaussian,
+    neg_log_likelihood_approx,
+)
+from ..utils.template_network_interpolator import (
+    DummyTemplateInterpolator,
+    DummyTimeInterpolator,
     TemplateNetworkInterpolator,
     TimeGradientInterpolator,
 )
+from .impact_utilities import (
+    EmptyImages,
+    create_seed,
+    guess_shower_depth,
+    rotate_translate,
+)
+from .reconstructor import (
+    HillasGeometryReconstructor,
+    InvalidWidthException,
+    ReconstructionProperty,
+    TooFewTelescopesException,
+)
 
-__all__ = ["ImPACTReconstructor", "energy_prior", "xmax_prior", "guess_shower_depth"]
-
-
-def guess_shower_depth(energy):
-    """
-    Simple estimation of depth of shower max based on the expected gamma-ray elongation
-    rate.
-
-    Parameters
-    ----------
-    energy: float
-        Energy of the shower in TeV
-
-    Returns
-    -------
-    float: Expected depth of shower maximum
-    """
-
-    x_max_exp = 300 + 93 * np.log10(energy)
-
-    return x_max_exp
-
+PROV = Provenance()
 
-def energy_prior(energy, index=-1):
-    return -2 * np.log(energy**index)
+INVALID_GEOMETRY = ReconstructedGeometryContainer(
+    telescopes=[],
+    prefix="ImPACTReconstructor",
+)
 
+INVALID_ENERGY = ReconstructedEnergyContainer(
+    prefix="ImPACTReconstructor",
+    telescopes=[],
+)
 
-def xmax_prior(energy, xmax, width=100):
-    x_max_exp = guess_shower_depth(energy)
-    diff = xmax - x_max_exp
-    return -2 * np.log(norm.pdf(diff / width))
+# These are settings for the iminuit minimizer
```

Comment:
I assume these are all fixed settings, i.e. things you don't ever expect the user to adjust?

If that is not the case, then you should add them to `__all__` and change the comments to use the form `#: Comment` so they become documented.